week-2-wide-perspective/w02e02-data-manipulation/src/orders/tiles/AddOrderButton.js

```diff
@@ -0,0 +1,17 @@
+import { faker } from '@faker-js/faker';
+import { axiosClient } from '../../utils/client';
+import { fromEvent } from 'rxjs';
+
+const button = document.querySelector(`[data-button="add-order"]`);
+
+button.addEventListener('click', async () => {
+  const { data = [] } = await axiosClient('/orders');
+  axiosClient.post('/orders/', {
+    id: faker.datatype.uuid(),
+    orderNumber: `${data.length + 1}/2022`,
+    orderDate: new Date(),
+    sale: faker.datatype.number(),
+  });
+});
+
+export const $click = fromEvent(button, 'click');
```

Comment:
Detal:
tutaj chyba chodzło o strumień klików więc: `click$` (ten dolar na koniec).

---

Ważniejsze:

Tutaj jest źródło wszystkich problemów Twojej aplikacji i konsekwencja późniejszych komplikacji.
Zauważ że średnio co 4 - 5 kliknięcie powoduje taki SKOK o `2` liczby załadowanych orders.

Dzieje się tak dlatego - że uzależniasz odświeżenie każdego z komponentów od kliknięcia w button a nie od "pojawienia się nowych danych"

Zobacz powyżej:
w momencie kliknięcia w button - wyślij `post` z `axiosClient` i Ty nie czekasz aż on "wróci" tylko od razu każdy kto subskrybuje kliknięcie - robi znów strzał do back-endu po dane.

robi się `racing condition` - czy pierwszy wykona się `POST` czy wróci `GET`.

Dodatkowo każdy z Twoich komponentów osobno znów - pyta o `/orders`
- czemu ?

Powinniśmy `RAZ` ściągnąć `/orders` i każdy kto subskrybuje strumień - powinien je dostać.
potem jeśli dodamy coś `POST` to znów pobieramy `/orders` i wtedy - każdy kto subskrybuje - odświeży.